### DISCUSSION TITLE
我发现用umi生成项目是typescript项目，我想用JavaScript来做React项目，请问下能不能通过Umi框架来生成？

### DISCUSSION THREAD

**Discussion by liaodalin19903** [umijs/umi#10419] [Q&A] [answered]
在阅读umi [getting-started](https://umijs.org/docs/tutorials/getting-started) 文档时候，
快速上手创建了umi项目，
```
$ npx create-umi@latest
Need to install the following packages:
  create-umi@latest
Ok to proceed? (y) y
✔ Pick Umi App Template › Simple App
✔ Pick Npm Client › npm
✔ Pick Npm Registry › taobao
Write: .gitignore
Write: .npmrc
Write: .umirc.ts
Write: package.json
Copy:  src/assets/yay.jpg
Copy:  src/layouts/index.less
Write: src/layouts/index.tsx
Copy:  src/pages/docs.tsx
Copy:  src/pages/index.tsx
Write: tsconfig.json
Copy:  typings.d.ts
 
> postinstall
> umi setup
```

但是我发现都是typescript项目，我想用JavaScript来做React项目，请问下能不能通过Umi框架命令来生成？

**Comment by fz6m** [comment, marked as answer]
在 umi 中的设计是支持 `js` + `jsx` / `ts` + `tsx` 来编写的，但：

1. 因为企业开发一直以 `ts` 为优先，所以没有对 `js` 的处理做详尽的测试，如使用 `js` 来编写项目可能存在 edge case 。

2. 社区近年来，“ts 优先“ 的思想更加普及，越来越多框架都是默认生成 ts 模板（如 nextjs 、react-native ），umi 是一个企业级项目的开源版本，所以学习并用 ts 编写任何项目是有绝对必要的。

3. 并不是绝对不存在 `js` 的情况，因为有一些上古项目还存在 `jsx` 编写的页面，他们可以和 `tsx` 混用。

**Reply by liaodalin19903** [reply]
是说当前安装快速开始创建的项目可以直接写jsx的文件吗？

**Reply by SilentFlute** [reply]
是的, 只是如作者所说, 可能会出现一些边界问题, 个人觉得如果项目不是太着急, 时间相对充裕, 建议直接上`ts`, 参照着`demo`来, 遇到不会的直接谷歌即可, 项目做完, `ts`也掌握得差不多了, 后续再使用`ts`就得心应手了
